dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -518,10 +584,11 @@ def process_data_chunk(self, sess, chunk_df):
             del offices
 
         # Gathering flex data (must be done before chunk limiting)
-        if self.reader.flex_fields:
-            flex_data = chunk_df.loc[:, list(self.reader.flex_fields + ['row_number'])]
-        if flex_data is not None and not flex_data.empty:
-            flex_data['concatted'] = flex_data.apply(lambda x: concat_flex(x), axis=1)
+        with lockable:
+            if shared_data['flex_fields']:
```

Comment:
Since this creates a lock around any processing of flex fields, it can create a bottleneck (sync point) for all chunks. It seems the only access to shared state is the bool flag for flex fields. Does this change on a per-chunk basis? Does anything during the processing of chunks write back to that value in the dict? If not, maybe take this out of the share state dict and just pass it in the function as a boolean, and that way no lock is needed.

Reply:
Good catch, makes sense, just tested it out, and it all works out.